x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/CompositeBucketsChangeCollector.java

```diff
@@ -425,14 +427,41 @@ public boolean queryForChanges() {
 
     static class HistogramFieldCollector implements FieldCollector {
 
+        // cutoff is calculated with max_range/current_range, current_range must be smaller
+        // the optimization gets only applied if we cut at least by 20%
```

Comment:
Just curious, why 20%?

Reply:
I made some benchmarks and saw that the new "optimized version" runs slower if the range does not cut anything(not dramatically slower, but slower). But I have no evidence that `20%` is the perfect value here. It's highly data dependent, probably there is no perfect answer, 20% seemed reasonable.